clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -147,6 +147,15 @@ struct BinOpInfo {
       return UnOp->getSubExpr()->getType()->isFixedPointType();
     return false;
   }
+
+  /// Does the BinaryOperator have the wraps attribute?
+  /// If so, we can ellide overflow sanitizer checks.
+  bool oneOfWraps() const {
+    const Type *TyPtr = E->getType().getTypePtrOrNull();
+    if (TyPtr)
```

Comment:
Also would be simplified with the above suggestion.  Also, since `QualType` has an operator -> and *, we typically don't fall to `Type*` unless we REALLY have to.

Reply:
Gotcha, checkout a2f63982920f22d795c4971800bcc5cb55356570